common/basecomponent/base.go

```diff
@@ -87,6 +89,7 @@ func NewBaseDendrite(cfg *config.Dendrite, componentName string) *BaseDendrite {
 		componentName: componentName,
 		tracerCloser:  closer,
 		Cfg:           cfg,
+		Cache:         caching.NewInMemoryLRUCache(), // TODO: make configurable
```

Comment:
I'd rather we hid the fact that we're using an LRU cache as an implementation detail, and not yank it up to the top unless we really feel we have to. We already have an LRU cache for the key database, and that puts the entire caching code in just one file: https://github.com/matrix-org/dendrite/blob/c019ad708669a4c269072bea933f0a520a78ca7c/common/keydb/sqlite3/server_key_table.go#L23 

I find this preferable as it's less stuff to worry about when perusing the codebase.

Reply:
The main motivation for taking it up to the `BaseComponent` was that it allows both the roomserver `api` and `query` packages to use the same cache when in monolith mode, rather than duplicating it. In polylith mode it will mean that each component predictably gets exactly one cache.

Reply:
Incidentally, I really don't like that the KeyDB has LRU caching as far down as the database-specific statement functions. At the deepest level it should be at the Database function layer, and probably sharing the same InMemoryLRUCache as room versions.